File: main_project/DistCache/distributedcache/src/main/java/xce/distcache/Distribution.java
package xce.distcache;

/**
 * 管理id->block的对应关系，固定关系。
 * 
 * */
public final class Distribution {
    public static final Distribution instance(int blockSize) {
	return new Distribution(blockSize);
    }

    private final int _blockSize;

    private Distribution(int blockSize) {
	if (blockSize < 1) {
	    blockSize = 1;
	}
	_blockSize = blockSize;
    }

    public int get(long id) {
	return (int) (Math.abs(id) % _blockSize);
    }
}

File: main_project/DistCache/distributedcache/src/main/cpp/Distribution.cpp
#include "Distribution.h"

using xce::distcache::DistributionPtr;
using xce::distcache::Distribution;

DistributionPtr Distribution::instance(int blockSize) {
	return DistributionPtr(new Distribution(blockSize));
}

Distribution::Distribution(int blockSize) {
	if (blockSize < 1) {
		blockSize = 1;
	}
	blockSize_ = blockSize;
}

int Distribution::get(long id) {
	return (int)(abs(id) % blockSize_);
}
